pkg/util/common.go

```diff
@@ -1089,8 +1088,8 @@ func GetRouteClient(restConfig *rest.Config, clientset *kubernetes.Clientset, na
 		return nil
 	}
 
-	ocVersion, err := GetOcVersion(clientset)
-	if err == nil && len(ocVersion) > 0 {
+	isOpenshift, err := IsOpenshift(clientset)
```

Comment:
Why don't we first check if we're on openshift, and then try to create a routeclient?

Reply:
+1 better to check outside this function and have GetRouteClient do 1 thing, which is return the route client.  Make the caller decide whether to use the function or not.